pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -834,6 +834,91 @@ int kretprobe__inet6_bind(struct pt_regs* ctx) {
     return sys_exit_bind(ret);
 }
 
+SEC("kprobe/sockfd_lookup_light")
+int kprobe__sockfd_lookup_light(struct pt_regs* ctx) {
+    int sockfd = (int)PT_REGS_PARM1(ctx);
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    bpf_map_update_elem(&sockfd_lookup_args, &pid_tgid, &sockfd, BPF_ANY);
+    return 0;
+}
+
+// this kretprobe is essentially creating an index of (PID, socketfd) to a conn_tuple_t object
+SEC("kretprobe/sockfd_lookup_light")
+int kretprobe__sockfd_lookup_light(struct pt_regs* ctx) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    int *sockfd = bpf_map_lookup_elem(&sockfd_lookup_args, &pid_tgid);
+    if (sockfd == NULL) {
+        return 0;
+    }
+
+    struct socket *socket = (struct socket*)PT_REGS_RC(ctx);
+    struct sock *skp;
+    bpf_probe_read(&skp, sizeof(skp), &socket->sk);
```

Comment:
Sadly, accessing the `sk` field of `struct socket` is not consistent across kernel versions.

Reply:
😭

Reply:
Actually, I may be wrong about this. It is a `struct file*` above, so that should be a constant size.

Reply:
@brycekahle this is not the kernel struct. is that still the case?

Reply:
`struct socket` is *not* the kernel struct?

Reply:
correct, it's the userspace one

Reply:
I extended the offset guessing to support `struct sock->sk` @brycekahle